diff --git a/js/hover_preview.js b/js/hover_preview.js
index 0d13ac9..03d26b9 100644
--- a/js/hover_preview.js
+++ b/js/hover_preview.js
@@ -3,14 +3,14 @@
  * Bootstrap Basic Image Gallery - hovering over thumbnails changes main.
  */
 
-(function ($, Drupal) {
+(function ($, Drupal, once) {
   'use strict';
 
   Drupal.behaviors.bootstrap_basic_image_gallery_hover_preview = {
     attach: function attach(context, settings) {
 
       // Find each thumbnail.
-      $.each($(context).find('.bootstrap-basic-image-gallery .thumb').once('hover_preview'), function (index, item) {
+      once('hover_preview', '.bootstrap-basic-image-gallery .thumb' , context).forEach((item, index) => {
         // On hover, exchange the main image with the thumbnail image.
         $(item).on('mouseover', function(ev) {
           // Get the thumbnail being hovered.
@@ -26,4 +26,4 @@
 
     }
   };
-})(jQuery, Drupal);
+})(jQuery, Drupal, once);
diff --git a/js/hover_preview_media.js b/js/hover_preview_media.js
index d999e7a..42c8e72 100644
--- a/js/hover_preview_media.js
+++ b/js/hover_preview_media.js
@@ -3,14 +3,14 @@
  * Bootstrap Basic Image Gallery - hovering over thumbnails changes main.
  */
 
-(function ($, Drupal) {
+(function ($, Drupal, once) {
   'use strict';
 
   Drupal.behaviors.bootstrap_basic_image_gallery_hover_preview = {
     attach: function attach(context, settings) {
 
       // Find each thumbnail.
-      $.each($(context).find('.bootstrap-basic-image-gallery .thumb').once('hover_preview'), function (index, item) {
+      once('hover_preview', '.bootstrap-basic-image-gallery .thumb' , context).forEach((item, index) => {
         // On hover, exchange the main image with the thumbnail image.
         $(item).on('mouseover', function(ev) {
           // Get the thumbnail being hovered.
@@ -30,4 +30,4 @@
       });
     }
   };
-})(jQuery, Drupal);
+})(jQuery, Drupal, once);
diff --git a/js/lazyload.js b/js/lazyload.js
index e1868ea..ebfc29f 100644
--- a/js/lazyload.js
+++ b/js/lazyload.js
@@ -10,7 +10,7 @@
     attach: function attach(context, settings) {
 
       // Find each lazy modal.
-      $.each($(context).find('.bootstrap-basic-image-gallery .modal.lazy').once('lazyload'), function (index, item) {
+      once('modal-lazyload', '.bootstrap-basic-image-gallery .modal.lazy' , context).forEach((item, index) => {
         // When the modal is shown, lazy load the active image.
         $(item).on('show.bs.modal', function(ev) {
           var slideToDelta = $(ev.relatedTarget).data('slide-to');
@@ -21,7 +21,7 @@
       });
 
       // Find each lazy carousel.
-      $.each($(context).find('.bootstrap-basic-image-gallery .carousel.lazy').once('lazyload'), function (index, item) {
+      once('carousel-lazyload', '.bootstrap-basic-image-gallery .carousel.lazy', context). forEach((item, index) =>{
         // When the carousel slides, lazy load the image.
         $(item).on('slide.bs.carousel', function(ev) {
           var lazyElement = $(ev.relatedTarget).find("img[data-src]");
