diff --git a/src/Plugin/Commerce/InlineForm/TipInlineForm.php b/src/Plugin/Commerce/InlineForm/TipInlineForm.php
index 5f19fd9..2f28d8b 100644
--- a/src/Plugin/Commerce/InlineForm/TipInlineForm.php
+++ b/src/Plugin/Commerce/InlineForm/TipInlineForm.php
@@ -145,7 +145,7 @@ class TipInlineForm extends InlineFormBase {
         '#type' => 'radios',
         '#options' => $tip_options,
         '#attributes' => [
-          'id' => 'tip_info_tip_checkboxes',
+          'class' => ['tip_info_tip_checkboxes'],
         ],
       ];
       $inline_form['tip_info']['other_tip'] = [
@@ -158,7 +158,7 @@ class TipInlineForm extends InlineFormBase {
         ],
         '#states' => [
           'visible' => [
-            ':input[id="tip_info_tip_checkboxes"]' => ['value' => 'other'],
+            ':input.tip_info_tip_checkboxes' => ['value' => 'other'],
           ],
         ],
         '#element_validate' => [
