diff --git a/src/Plugin/Commerce/CheckoutPane/CustomCheckoutMessage.php b/src/Plugin/Commerce/CheckoutPane/CustomCheckoutMessage.php
index c8f0f5206c6e4979af49cf587e370d2021cb492b..e648645a70fe1d61ff5993e610c6a05674ba95dd 100644
--- a/src/Plugin/Commerce/CheckoutPane/CustomCheckoutMessage.php
+++ b/src/Plugin/Commerce/CheckoutPane/CustomCheckoutMessage.php
@@ -55,7 +55,7 @@ class CustomCheckoutMessage extends CheckoutPaneBase {
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, CheckoutFlowInterface $checkout_flow = NULL) {
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, ?CheckoutFlowInterface $checkout_flow = NULL) {
     return new static(
       $configuration,
       $plugin_id,
