diff --git a/a/src/Plugin/views/filter/SimplenewsStatsAction.php b/b/src/Plugin/views/filter/SimplenewsStatsAction.php
index a75489d..87630ef 100644
--- a/a/src/Plugin/views/filter/SimplenewsStatsAction.php
+++ b/b/src/Plugin/views/filter/SimplenewsStatsAction.php
@@ -38,6 +38,7 @@ class SimplenewsStatsAction extends FilterPluginBase {
         'click' => $this->t('Click'),
         'view' => $this->t('View'),
         'sent' => $this->t('Sent'),
+        'unsubscribe' => $this->t('Unsubscribe'),
       ],
     ];
   }
diff --git a/a/src/SimplenewsStatsPage.php b/b/src/SimplenewsStatsPage.php
index 089e9ac..829cf54 100644
--- a/a/src/SimplenewsStatsPage.php
+++ b/b/src/SimplenewsStatsPage.php
@@ -71,6 +71,14 @@ class SimplenewsStatsPage {
    */
   protected ?int $countView = NULL;
 
+  /**
+   * Number of unsubscribes.
+   *
+   * @var int|null
+   * @see queryCount()
+   */
+  protected ?int $countUnsubscribe = NULL;
+
   /**
    * SimplenewsStatsPage Constructor.
    *
@@ -119,6 +127,16 @@ class SimplenewsStatsPage {
     return $this->queryCount('view');
   }
 
+  /**
+   * Return the total of Unsubscribes.
+   *
+   * @return int
+   *   The number of unsubscribes.
+   */
+  public function getCountUnsubscribes(): int {
+    return $this->queryCount('unsubscribe');
+  }
+
   /**
    * Return the total of Views.
    *
@@ -162,6 +180,22 @@ class SimplenewsStatsPage {
     ];
   }
 
+  /**
+   * Return the detail of unsubscribe actions.
+   *
+   * @return array
+   *   The detail of unsubscribes formatted for chart.
+   */
+  public function getDetailUnsubscribes(): array {
+    return [
+      'label' => $this->t('Unsubscribes'),
+      'backgroundColor' => '#ff6384',
+      'borderColor' => '#ff6384',
+      'fill' => FALSE,
+      'data' => $this->queryDetail('unsubscribe'),
+    ];
+  }
+
   /**
    * Calculation of percent.
    *
@@ -256,9 +290,11 @@ class SimplenewsStatsPage {
         $this->t('Sent status'),
         $this->t('Views'),
         $this->t('Clicks'),
+        $this->t('Unsubscribes'),
         $this->t('Total emails sent'),
         $this->t('% Views'),
         $this->t('% Clicks'),
+        $this->t('% Unsubscribes'),
         $this->t('Detail'),
       ],
       '#rows' => [
@@ -268,9 +304,11 @@ class SimplenewsStatsPage {
           $this->simplenewsValues['status'],
           $this->getCountViews(),
           $this->getCountClicks(),
+          $this->getCountUnsubscribes(),
           $this->getCountTotalMails(),
           $this->getPercent($this->getCountViews()),
           $this->getPercent($this->getCountClicks()),
+          $this->getPercent($this->getCountUnsubscribes()),
           $this->getLinkDetail(),
         ],
       ],
@@ -435,6 +473,7 @@ class SimplenewsStatsPage {
     if (empty($this->series)) {
       $this->series[] = $this->getDetailClicks();
       $this->series[] = $this->getDetailViews();
+      $this->series[] = $this->getDetailUnsubscribes();
     }
     return $this->series;
   }
